GOFevaluation/utils.py

```diff
@@ -136,7 +135,29 @@ def _get_count_density(ns, be_first, be_second, data_sample):
     return ns
 
 
-def _get_equiprobable_binning(reference_sample, n_partitions, order=None):
+def _weighted_equi(n_bins, data, weights):
+    """Perform a 1D equiprobable binning for the reference sample. The binning
+    is defined such that the number of counts in each bin are (almost) equal.
+    Bins are defined based on the ECDF of the reference sample.
+    :param n_bins: number of partitions in this dimension
+    :type n_bins: int
+    :param data: sample of unbinned reference
+    :type data: array_like, 1-Dimensional
+    :param weights: weights of samples
+    :type weights: array_like, 1-Dimensional
+    :return: Returns bin_edges.
+    :rtype: array_like, 1-Dimensional
+    """
+    weights = weights[data.argsort()]
+    data = data[data.argsort()]
+    cumsum = np.cumsum(weights)
+    cumsum -= cumsum[0]
+    bin_edges = np.interp(np.linspace(0, 1, n_bins + 1)[1:-1], cumsum / cumsum[-1], data)
```

Comment:
The "old" method basically only used `np.percentile` under the hood. I would expect this to be quite a bit faster than `np.interp`. Have you checked the speed of this old vs. new? If the change in speed is marginal, I don't mind it but otherwise, I would suggest doing something as fast as possible in the case that all weights are one. What do you think?

Reply:
The original [numpy.percentile](https://numpy.org/doc/stable/reference/generated/numpy.percentile.html) will not work here because it is essentially a map between CDF of weights and `reference_sample`. But I will switch to quantile when weights are not provided for performance concerns.